src/components/hintSrv/HintSrv.js

```diff
@@ -0,0 +1,99 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.hint').provider('HintSrv',function(){
+        var registeredHints = {};
+
+        this.registerHint = function (hintName, hintAction, determineWhetherToTriggerFnGetter) {
```

Comment:
Add a map which will hold all the hint names, this map should be exposed by the service and when triggering hint the name should be  retrieved from this map.